Resources/Prototypes/Entities/Objects/Weapons/Melee/e_sword.yml

```diff
@@ -0,0 +1,18 @@
+- type: entity
+  name: energy sword
+  parent: BaseItem
+  id: EnergySword
+  description: A very dangerous energy sword. Can be stored in pockets when turned off. Makes a lot of noise when used or turned on.
+  components:
+  - type: Esword
+  - type: Sprite
+    sprite: Objects/Weapons/Melee/e_sword.rsi
+    state: e_sword
```

Comment:
For two different reasons you should probably split the "sword hilt" and the "blade" into separate layers on the sprites.

Firstly, you can then make the blade-layer unshaded (i.e., ignore lights). This is used by computer screens so they are visible in the dark, even without needing a point light. Just look for `shader: unshaded` in the various yaml files for examples.

Secondly, it will then make it easier to add coloured swords, because you could just use something like 
```cs
var color = Color.Red // yaml datafield
sprite.LayerSetColor(bladeLayer, color);
item.Color = color; // in hand sprite
``` 
If you do this currently, it also colours the handle, because the handle and the sword itself are on the same layer.

Reply:
Actually, if you did that you could just get the rainbow effect working by just adding a `RgbLightControllerComponent` and tweaking `RgbLightControllerSystem` so it also updates item in-hands, and can be selective about the sprite layers.